### PR TITLE
feat(test): test updates and deletes for backwards compatibility

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

#9973 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by huangjw806 — APPROVED** [review]
LGTM!

**Review by fuyufjh — APPROVED** [review]
LGTM

File: ci/scripts/backwards-compat-test.sh [hunk 1]
```diff
@@ -63,14 +69,69 @@ assert_eq() {
...
+  seed_table 1 10000
+
+  run_sql "CREATE MATERIALIZED VIEW m as SELECT * from t;" &
+  CREATE_MV_PID=$!
+
+  seed_table 10001 20000
+
+  random_update 1 20000 1000
+
+  random_delete 1 20000 1000
```
Comment (lmatz) [inline ci/scripts/backwards-compat-test.sh:122]: I was thinking if the `delete`s should happen after we start the new cluster,
so we test if the new RW can successfully delete the rows that were ingested by the old cluster.

**Review by lmatz — APPROVED** [review]
rest LGTM

File: ci/scripts/backwards-compat-test.sh (hunk 1 above)
Comment (kwannoel) [reply ci/scripts/backwards-compat-test.sh:122]: Oh I see. I will add this as well.

Comment (kwannoel) [reply ci/scripts/backwards-compat-test.sh:126]: Tested here: https://github.com/risingwavelabs/risingwave/pull/10785/commits/131494e12ef8226a99b386c65adfc0edad2282ab#diff-2cd274f1db89f49e0ff90c98f5b8cbbd66241f386285b2193b5121d5c4aacbeaR261.

Just run updates and deletes after new cluster started. If no errors, test pass.

**Pull request merged**